### PR TITLE
Convert to a ES Module (blocked on next-major)

### PR DESCRIPTION
**Pull request by koddsson**
Describe your changes here.

Convert `primer/react` to a ES Module.

### Screenshots

N/A

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Tested in Chrome
- [x] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge

### CODE REVIEW

File: src/__tests__/__snapshots__/themePreval.test.ts.snap [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
 
 exports[`snapshot theme-preval.js 1`] = `
-"// this file was prevaled
+"var module = {
+  exports: {}
+};
+var exports = module.exports;
+// this file was prevaled
 // This file needs to be a JavaScript file using CommonJS to be compatible with preval
 // Cache bust: 2022-07-25 12:00:00 GMT (This file is cached by our deployment tooling, update this timestamp to rebuild this file)
```
Comment (koddsson) [inline src/__tests__/__snapshots__/themePreval.test.ts.snap:10]: I don't know what `preval` is so I don't know what to do about this.

File: script/build [hunk 2]
```diff
@@ -5,17 +5,15 @@ set -e
 # Clean up
 rm -rf dist
 rm -rf lib
-rm -rf lib-esm
 
 # Bundle
 npx cross-env NODE_ENV=production rollup -c
 
 # Compile
-npx cross-env BABEL_MODULE=commonjs babel src --out-dir lib --extensions ".js,.ts,.jsx,.tsx"
```
Comment (koddsson) [inline script/build:14]: I don't think we need a commonjs build so I removed it.

**Comment by joshblack** [comment]
Hey @koddsson! 👋 Wanted to follow-up on this real quick 👀 

It seems like this kind of change will need to be slated to the next major release (v36) and I've added it over to the proposals here: https://github.com/github/primer/issues/1294

As a result, it might be better to revisit it closer to the next major release as I'm not sure if changes can be merged before then.

Hope this makes sense! Let me know if you have any questions

**Comment by joshblack** [comment]
Hi there! 👋 Just wanted to say thanks again for the taking the time to work on this, we picked it up and are building on it over in: https://github.com/primer/react/pull/3410

Going to close this out as a result, let me know if you have any questions! Thanks again 🙏

**Pull request closed without merging**